onnxruntime/core/session/inference_session.cc

```diff
@@ -1024,16 +1024,21 @@ Status InferenceSession::LoadOrtModelWithLoader(std::function<Status()> load_ort
   const auto* fbs_ort_model_version = fbs_session->ort_version();
   ORT_RETURN_IF(fbs_ort_model_version == nullptr, "Serialized version info is null. Invalid ORT format model.");
 
+  // if this is a full build we can update an old format model.
+  bool model_needs_update = IsOrtModelVersionSupported(fbs_ort_model_version->string_view()) == false;
+
+#if defined(ORT_MINIMAL_BUILD)
   // Note about the ORT format version 5 breaking change.
   // TODO This change was introduced in 1.13. Remove this note a few releases later, e.g., 1.15.
   constexpr auto* kOrtFormatVersion5BreakingChangeNote =
       "This build doesn't support ORT format models older than version 5. "
       "See: https://github.com/microsoft/onnxruntime/blob/rel-1.13.0/docs/ORT_Format_Update_in_1.13.md";
```

Comment:
if this change makes it into 1.13, it might be good to update this doc to mention the possibility of upgrading an old ORT format model with a full build.

Reply:
Probably won't make 1.13 but upgrade could be done with a nightly build soon. Can update the doco to mention that once we have a nightly that includes this change.